src/HaskellCI.hs

```diff
@@ -302,7 +305,91 @@ regenerateBash opts = do
             doBash argv f ( optionsWithOutputFile fp <> opts' <> opts)
   where
     noBashScript :: IO ()
-    noBashScript = putStrLn "No haskell-ci.sh, skipping bash regeration"
+    noBashScript = putStrLn "No haskell-ci.sh, skipping bash regeneration"
+
+-------------------------------------------------------------------------------
+-- GitHub actions
+-------------------------------------------------------------------------------
+
+defaultGitHubPath :: FilePath
+defaultGitHubPath = ".github/workflows/haskell-ci.yml"
+
+doGitHub :: [String] -> FilePath -> Options -> IO ()
+doGitHub args path opts = do
+    contents <- githubFromConfigFile args opts path
+    case optOutput opts of
+        Nothing              -> do
+            createDir defaultGitHubPath
```

Comment:
The code duplication among all these backends is a bit worrisome, especially since there are now subtle differences between backends (e.g., the GHA backend uses `createDir`+`writeFile`, while other backends just use `writeFile`). Is there a compelling reason not to factor out the code that is shared among all the backends?

Reply:
I don't find it yet problematic. The code duplication in actual generation is something which we need to dedup, but there as well we should do it only when we feel the pain, not preemptively. My previous attempts to introduce extra backends failed because I tried to generalize to early.